### PR TITLE
Pull Req: allow developer to specify splash screen

### PR DESCRIPTION
**Pull request by rwmtse**
-Implements feature blackberry/BB10-Webworks-Packager#109
-Unit test included

### CODE REVIEW

**Comment by rwmtse** [comment]
Thanks @jkeshavarzi. I will move the validation to validateConfig.

File: test/unit/lib/config-parser.js [hunk 1]
```diff
@@ -378,4 +378,26 @@ describe("xml parser", function () {
...
     });
+
+    it("does not throw error when a valid splash screen image is specified", function () {
+        var data = testUtilities.cloneObj(testData.xml2jsConfig);
+        data["rim:splashscreen"] = "test.JPG:anything.png";
+
+        mockParsing(data);
+
+        expect(function () {
+            configParser.parse(configPath, session, function (configObj) {});
```
Comment (jamesjhedley) [inline test/unit/lib/config-parser.js]: I think we should have a test that checks that the actual splashScreen property was set correctly during parsing.
Something like expect(configObj.splashScreen).toEqual("test.JPG")

Comment (rwmtse) [inline test/unit/lib/config-parser.js]: Added, see line 390

**Comment by jamesjhedley** [comment]
Only other thing i might do is make config.splashScreen an array.Not a big deal though.

Code review complete.

**Comment by rwmtse** [comment]
The reason why it's not an array is because config.splashscreen will be passed as-is to blackberry-tablet.xml, that is the format native application descriptor expects.

**Comment by cdelcol** [comment]
i think we need to be careful with the config.xml design here. Let's say BB10 makes it onto the tablet, what does my config look like. I don't really want to put the onus on the dev to re-package with a different config.xml.

something like this?

``` XML
<rim:splashscreen>
<BB10-Colt>
<...stuff here.../>
</BB10-Colt>
</rim:spalshscreen>
```

We will run into this with icons, possibly. Where else?

Questions:
Does the manifest handle all this for us?
Does the native SDK assume per-device .bar files?

I want to make sure we don't design ourselves into a corner here.

**Comment by cdelcol** [comment]
As well, should we start to reserve a directory for all of the app "meta assets"

/<reserved_dir>/icons
/<reserved_dir>/splashscreen/

That would allow the config to be much cleaner, yet the dev must know where to put things, and possibly name them.

I'm not saying any of this is the best approach, but decisions made for the humble splash screen will have a ripple effect through other features.

**Comment by rwmtse** [comment]
The manifest currently supports just one screen resolution, pretty restricted:
https://bdsc.webapps.blackberry.com/native/documentation/com.qnx.doc.native_sdk.devguide/com.qnx.doc.native_sdk.devguide/topic/r_barfile_dtd_ref_splashscreen.html

Let me dig into their docs further

**Comment by rwmtse** [comment]
@cdelcol See the discussion with Ken here: blackberry/BB10-Webworks-Packager#109

**Comment by cdelcol** [comment]
i don't think we can just ignore the internationalization. At a minimum we should open up a backlog item to support it. I have alot of meetings with those teams, and it will likely all need to get done before final GC.

Rather than relying on current state, has anyone tried to see how this will be handled in future releases? this must be designed internally already.

I'm ok with doing a version 1 and changing/adding mid-summer. But I really don't like to mess with the config.xml once shipped. Changing api signatures (ie code not working right) is one thing after a beta. But when an app doesn't package? a bit harsh.

**Comment by nukulb** [comment]
Let us ignore localization/internationalization for now - spawned a new feature that will deal with internationalization for config.xml elements - blackberry/BB10-Webworks-Packager#122

@kwallis - please approve/change the acceptance criteria for blackberry/BB10-Webworks-Packager#122 and attach milestone information to this please.

**Comment by nukulb** [comment]
@kwallis- multiple screen resolutions are still an issue.

**Comment by rwmtse** [comment]
I have gotten some answers about the plan to support multiple resolutions. Still waiting for more answers, will post here when I get something more concrete.

**Comment by ishneur** [comment]
Testing completed on Windows. Opened the following issues:
https://github.com/blackberry/BB10-Webworks-Packager/issues/121
https://github.com/blackberry/BB10-Webworks-Packager/issues/123

There are a few more issues:
- If I specify more than two images for the splash screen (`<rim:splashscreen>`splash-landscape.jpg:splash-portrait.jpg:splash-portrait.png`</rim:splashscreen>`), then the portrait screen is displayed as a blank white page
- When the first image is not specified (`<rim:splashscreen>`:splash-portrait.jpg`</rim:splashscreen>`), then the landscape screen is displayed as a blank white page

Since the markup is going to change, some of those issues might not be applicable anymore and can be retested later on.

I'll also do the testing on Mac and post the results once I'm done.

**Comment by ishneur** [comment]
Testing completed on Mac. Same behavior as on Windows.

**Comment by nukulb** [comment]
Closing the Pull Request - NOT TO BE MERGED YET - We are not sure if this will make it to Beta2 milestone or not.

**Pull request closed without merging**